### PR TITLE
Load memo v2 into genesis for test validator

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Memo 2 is not available in the solana-test-validator

#### Summary of Changes
- Load memo 2 into test validator genesis
- Update program test to use memo v2
- Add label in explorer for memo 2

Fixes #

### CODE REVIEW

File: fetch-spl.sh [hunk 1]
```diff
@@ -40,6 +40,7 @@ fetch_program() {
 
 fetch_program token 3.1.0 TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA BPFLoader2111111111111111111111111111111111
 fetch_program memo  1.0.0 Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo BPFLoader1111111111111111111111111111111111
```
Comment (mvines) [inline fetch-spl.sh:42]:
```suggestion
```

Comment (jstarry) [reply fetch-spl.sh:42]: backwards incompatible!

Comment (jstarry) [reply fetch-spl.sh:42]: I don't mind tossing it, just didn't think it would do any harm.

Comment (mvines) [reply fetch-spl.sh:42]: actually yeah that was a bad idea.  But `program-test/src/programs.rs` should then include both memo programs too right?

Comment (jstarry) [reply fetch-spl.sh:42]: Yeah, that was my laziness. I'll add v1 / v2 modules

File: program-test/src/programs.rs [hunk 2]
```diff
@@ -4,15 +4,15 @@ mod spl_token {
...
+    solana_sdk::declare_id!("MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr");
 }
 mod spl_associated_token_account {
     solana_sdk::declare_id!("ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL");
 }
 
 static SPL_PROGRAMS: &[(Pubkey, &[u8])] = &[
     (spl_token::ID, include_bytes!("programs/spl_token-3.1.0.so")),
-    (spl_memo::ID, include_bytes!("programs/spl_memo-1.0.0.so")),
+    (spl_memo::ID, include_bytes!("programs/spl_memo-3.0.0.so")),
```
Comment (mvines) [inline program-test/src/programs.rs:15]: The new SO needs to be added to the repo as well in here: https://github.com/solana-labs/solana/tree/master/program-test/src/programs

Comment (jstarry) [reply program-test/src/programs.rs:15]: Ah thanks, for some reason I thought this picked up from fetch-spl

Comment (mvines) [reply program-test/src/programs.rs:15]: Would be nice :)
The problem is when the program-test crate is published to crates.io, ` fetch-spl.sh` is no longer a thing

File: fetch-spl.sh (hunk 1 above)
Comment (jstarry) [reply fetch-spl.sh:42]: To be clear, this memo is v2 despite the 3.0.0 semver right?

Comment (mvines) [reply fetch-spl.sh:42]: It's the second memo program we've published to mainnet, yes.  Using spl_memo_1_0 and spl_memo_3_0 is probably the least confusing from a Cargo.toml/program.rs perspective

Comment (jstarry) [reply fetch-spl.sh:42]: sounds good, thanks

**Review by mvines — APPROVED** [review]
lgtm beyond program-test/src/programs/spl_memo-3.0.0.so being empty

**Comment by jstarry** [comment]
> lgtm beyond program-test/src/programs/spl_memo-3.0.0.so being empty

🤕

**Comment by jstarry** [comment]
The file appears to be non-empty based on this commit diff https://github.com/solana-labs/solana/pull/15425/commits/b314b13566e30d04c7d177d44a279e14a90b7633

**Comment by jstarry** [comment]
ok now that diff shows +0... it was *just* showing +73KB

**Comment by jstarry** [comment]
Screen Shot 2021-02-19 at 2 26 22 PM
bizarre

**Comment by jstarry** [comment]
Merging because when I view the file, it shows 73 KB.

**Pull request merged**